tests/codegen-units/item-collection/cross-crate-closures.rs

```diff
@@ -3,14 +3,14 @@
 //@ compile-flags:-Zprint-mono-items=eager -Zinline-mir=no
 
 #![deny(dead_code)]
-#![feature(start)]
+#![crate_type = "lib"]
```

Comment:
Root collection has this extra code for bin crates that the test suite should be exercising: https://github.com/rust-lang/rust/blob/f5079d00e63718235f95353df38ebb0891bab5b4/compiler/rustc_monomorphize/src/collector.rs#L1487

But I think this test suite isn't worth any additional effort, and perhaps only merits an issue.

Reply:
Some tests changed slightly, maybe that is related to that...

Reply:
Hm, no... making this a binary again with a bare `main` function still makes things `External`.

Reply:
I really don't think that this test suite is worth much of anyone's time. `-Zprint-mono-items=eager` changes the behavior of MonoItem collection in a way that's otherwise only used by `-Clink-dead-code`, but without the other effects of `-Clink-dead-code`. In short, over-fitting to the behavior of `-Zprint-mono-items=eager` (which is what these tests do by also printing the MonoItem Linkage) is testing a configuration nobody uses.